### PR TITLE
Presentation: `CompareDateTime` floating point precision fix

### PR DESCRIPTION
**Pull request by MartynasStrazdas**
Closes https://github.com/iTwin/itwinjs-backlog/issues/574

### CODE REVIEW

File: iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp [hunk 1]
```diff
@@ -2031,9 +2031,22 @@ struct ECExpressionToECSqlConverter : NodeVisitor
...
+            m_ignoredNodes.insert(node.GetRightCP());
+            return WrapPreviousNode("ABS", "< (1.0 / 86400000)");
+            }
+        if (left.StartsWith("0") && right.StartsWith("CompareDateTimes"))
+            {
+            m_nodesStack.pop_back();
+            m_ecsql = m_ecsql.substr(0, m_ecsql.length() - 1);
+            Append("(1.0 / 86400000) >");
+            return Append("ABS");
+            }
```
Comment (saskliutas) [inline iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp:2047]: I think this can be extracted into separate function

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp [hunk 2]
```diff
@@ -2029,11 +2029,35 @@ struct ECExpressionToECSqlConverter : NodeVisitor
...
+    /*-----------------------------------------------------------------------------**//**
+    * @bsimethod
+    +---------------+---------------+---------------+---------------+---------------+--*/
+    void HandleEqualtyNode(ComparisonNodeCR node)
+        {
         Utf8String left = node.GetLeftCP()->ToExpressionString();
+        Utf8String right = node.GetRightCP()->ToExpressionString();
+        if (TryHandleCompareDateTimeEquality(node, left, right))
+            return;
+        Append(node.ToString());
```
Comment (saskliutas) [inline iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp:2060]:
```suggestion
        if (TryHandleCompareDateTimeEquality(node))
            return;
        Append(node.ToString());
        Utf8String left = node.GetLeftCP()->ToExpressionString();
```

File: iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp [hunk 3]
```diff
@@ -2029,11 +2029,35 @@ struct ECExpressionToECSqlConverter : NodeVisitor
     /*-----------------------------------------------------------------------------**//**
     * @bsimethod
     +---------------+---------------+---------------+---------------+---------------+--*/
-    void HandleEqualtyNode(ComparisonNodeCR node)
+    bool TryHandleCompareDateTimeEquality(ComparisonNodeCR node, Utf8String left, Utf8String right)
         {
-        Append(node.ToString());
+        if (left.StartsWith("CompareDateTimes") && right.StartsWith("0"))
```
Comment (saskliutas) [inline iModelCore/ECPresentation/Source/Shared/ECExpressions/ECExpressionContextsProvider.cpp:2034]: How this will work with `0.5`?

**Review by grigasp — CHANGES_REQUESTED** [review]
Will this handle non-equality operators?
- `CompareDateTimes(this.PropertyName, 2) < 0`
- `CompareDateTimes(this.PropertyName, 2) <= 0`
- `CompareDateTimes(this.PropertyName, 2) > 0`
- `CompareDateTimes(this.PropertyName, 2) >= 0`

**Review by grigasp — APPROVED** [review]

**Review by saskliutas — APPROVED** [review]

**Pull request merged**